README.md

```diff
@@ -7,6 +7,50 @@ CSearch
 Making Code Search Great Again.
 
 
+Developing CSearch
+------------------
+Install the [Rust] development tools on your system with [rustup] if they are
+not already available. Then build and test the project using:
+
+    cargo test
+
+[Rust]: https://www.rust-lang.org
+[rustup]: https://www.rustup.rs
+
+
+Docker
+------
+A [Docker] container definition is provided with installations of the tools
+used to develop CSearch. To use the container, first install Docker if not
+already available and start a Docker terminal. Then create the container by
+running the following build at the top level of the repository source tree:
+
+    docker build --rm=true -t csearch .
+
+[Docker]: http://docker.io
+
+Once built, an interactive shell can be run in the container using:
+
+    docker run -it -v "$(pwd):/csearch" --workdir=/csearch csearch /bin/bash
+
+The current working directory from the host machine is available as the current
+directory in the container so it is possible to build and test the library as
+described earlier.
+
+    cargo test
+
+
+Running Clippy Lints
+--------------------
+[Clippy] is a Rust linter. Currently it has to be run manually since CSearch
+targets Rust stable and Clippy requires Rust nightly. Switching versions is
+easy with `rustup` - use the following to lint the repository:
+
+    rustup run nightly cargo clippy
```

Comment:
Do you think it would make sense to have the Dockerfile install nightly by default?